src/visitable.cpp

```diff
@@ -364,10 +364,8 @@ static VisitResponse visit_internal( const std::function<VisitResponse( item *,
                 return VisitResponse::NEXT;
             }
 
-            for( auto &e : node->contents ) {
-                if( visit_internal( func, &e, node ) == VisitResponse::ABORT ) {
-                    return VisitResponse::ABORT;
-                }
+            if( node->contents.visit_contents( func, parent ) == VisitResponse::ABORT ) {
```

Comment:
This uses the parent of the item that contains the `item_contents` object as parameter. But for the items inside the `item_contents` object, this is more like the grandparent.

In other words: `visit_contents` has no way of knowing the item that it is contained in.

I suggest the function should take the object containing the `items_content` object as parent. (This is also what the current code does: it forwards `node` and not `parent`.)